cni/pkg/ambient/podutil.go

```diff
@@ -65,7 +65,11 @@ func podHasSidecar(pod *corev1.Pod) bool {
 }
 
 func ztunnelPod(pod *corev1.Pod) bool {
-	return pod.GetLabels()["app"] == "ztunnel"
+	// Short term fix - we may make it customisable later
+	if pod.Namespace == "istio-system" || pod.Namespace == "kube-system" {
```

Comment:
You can use SystemNamespace here, and i am against kube-system again here

Reply:
+1 - we should be decreasing our use of hardcoded istio-system references; not increasing it

Reply:
Hmm that still doesn't feel quite right; wouldn't that mean that ztunnel could be any any namespace that the CNI excludes? That feels too lose

Reply:
In istiod CA we do ` CA_TRUSTED_NODE_ACCOUNTS: "istio-system/ztunnel,kube-system/ztunnel"`.

Which is based on SA -- not label. maybe that would work?

Reply:
Service account is certainly a stronger check...do we anticipate any perf impact?

Reply:
```
Hmm that still doesn't feel quite right; wouldn't that mean that ztunnel could be any any namespace that the CNI excludes? That feels too loose
```

Oh my bad, I misread the intent of this change. Correct.

Why check the namespace at all, then?

Reply:
The intent is to not allow some random developer in 'evil-namespace' to create a pod with the label app=ztunnel which would suddenly get all the traffic on the node. 

The CA_TRUSTED_NODE_ACCOUNTS also have istio-system and kube-system hardcoded - we can parse it to extract the list. We could also check the ztunnel is the service account - but it doesn't help since a malicious (or accidental) user can also create ztunnel SA in evil-namespace.

Reply:
the CA_TRUSTED_NODE_ACCOUNTS prevents " create ztunnel SA in evil-namespace.". It takes SA+NS into account

Reply:
@hzxuzhonghu - the choice of kube-system and istio-system is to be consistent with the CA_TRUSTED_NODE_ACCOUNTS ( which is a setting on Istiod - not CNI ) and with the practice to run it in one of the 2 namespaces.

On GKE we recommend kube-system today, and most CNI implementations are in kube-system.

I can make it a constant or flag - but default  values should be consistent.

Reply:
> Service account is certainly a stronger check...do we anticipate any perf impact?

If you have permission to deploy in a namespace - you can use any SA in that namespace. Yes, in theory some admission controller may check that - but once you have deploy permissions in kube-system or istio-system I don't think you should put too much trust in any admission controller.

The namespace is the main security boundary in K8S - and both kube-system and istio-system are very security critical and should be equally restricted.

Reply:
I'm a -1 on hardcoding `istio-system` on principle and practically. From a selfish practical perspective, my company's distribution uses a separate namespace (`aks-istio-system`) for Istio installation. From a principle perspective, the name of the "system" namespace for a piece of software should be inconsequential. Users, vendors, whomever should be able to change it if they wish. Defaults are fine and sensible, but we should break the assumption that components are going to live in `istio-system`

Reply:
`NODEPROXY_VALID_NS` as either a CNI env var (ideally templatable with Helm) or in the CNI config map (which I think only the plugin installer looks at but which the daemonset should be perfectly capable of reading, also templatable with Helm) makes sense to me.